src/main/java/com/timgroup/statsd/ServiceCheck.java

```diff
@@ -0,0 +1,166 @@
+package com.timgroup.statsd;
+
+/**
+ * A service check model, which is used to format a service check message
+ * sent to the datadog agent
+ */
+public class ServiceCheck {
+
+    private String name, hostname, message;
+
+    private int status, checkRunId, timestamp;
+
+    private String[] tags;
+
+    /**
+     */
+    public ServiceCheck() {
+    }
+
+    /**
+     * @param name
+     * @param status
+     */
+    public ServiceCheck(String name, int status) {
+            this.name = name;
+            this.status = status;
+    }
+
+    /**
+     * @return
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * @param name
+     */
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    /**
+     * @return
+     */
+    public int getStatus() {
+        return status;
+    }
+
+    /**
+     * @param status
+     */
+    public void setStatus(int status) {
+        this.status = status;
+    }
+
+    /**
+     * @return
+     */
+    public String getMessage() {
+        return message;
+    }
+
+    /**
+     * @return
+     */
+    public String getEscapedMessage() {
+        return message.replace("\n", "\\n").replace("m:", "m\\:");
+    }
+
+    /**
+     *
+     * @param message
+     */
+    public void setMessage(String message) {
+        this.message = message;
+    }
+
+    /**
+     * @return
+     */
+    public String getHostname() {
+        return hostname;
+    }
+
+    /**
+     * @param hostname
+     */
+    public void setHostname(String hostname) {
+        this.hostname = hostname;
+    }
+
+    /**
+     * @return
+     */
+    public int checkRunId() {
```

Comment:
Hmm, I thought I had ported that from elsewhere, but I can't seem to find where it came from. I'll remove it here and in the dd-agent PR.